platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -89,21 +84,7 @@ public Hash(@NonNull final byte[] value, @NonNull final DigestType digestType) {
         }
 
         this.digestType = digestType;
-        this.value = value;
-        this.bytes = Bytes.wrap(this.value);
-    }
-
-    /**
-     * Instantiate a hash with a byte buffer and a digest type.
-     *
-     * @param byteBuffer
-     * 		a buffer that contains the data for this hash
-     * @param digestType
-     * 		the digest type of this hash
-     */
-    public Hash(@NonNull final ByteBuffer byteBuffer, @NonNull final DigestType digestType) {
-        this(digestType);
-        byteBuffer.get(this.value);
+        this.bytes = Bytes.wrap(value);
```

Comment:
Note that this relies on the calling code to be nice and never modify the bytes in the array. I'm not sure about the performance implications - perhaps they outweigh the concern - but I think that making a copy of the array here would be safer from immutability perspective. Otherwise, I suggest to update the javadoc for this constructor (and perhaps this class, too) to clarify the requirement for the callers to be nice.

Reply:
+1

Reply:
I would not introduce the performance overhead of copying the array, I will add comments.

Reply:
In the future, we can move towards not using `byte[]` at all

Reply:
Sounds good.